python/ql/src/experimental/Security/CWE-295/UrlopenWithoutValidation.ql

```diff
@@ -0,0 +1,45 @@
+/**
+ * @name Attempt to open url without certificate validation
+ * @description Opening a url without certificate validation can allow man-in-the-middle attacks.
+ * @kind problem
+ * @problem.severity error
+ * @id py/urlopen-without-cert-validation
+ * @tags security
+ *       external/cwe/cwe-295
+ */
+
+import python
+import semmle.python.ApiGraphs
+
+predicate calls_urllib(CallNode call) {
+  call =
+    API::moduleImport(["urllib", "urllib2"])
+        .getMember("request")
+        .getMember("urlopen")
+        .getACall()
+        .getNode()
+}
+
+predicate is_https(CallNode call) {
+  exists(ControlFlowNode http_arg, StringValue http_string |
+    http_arg = call.getArg(0) and
+    http_arg.pointsTo(http_string) and
+    http_string.getText().matches("https://%")
+  )
+}
```

Comment:
```suggestion
predicate is_https(DataFlow::CallCfgNode call) {
  call.getArg(0).getALocalSource().asExpr() = any(StrConst s | s.getText().matches("https://%"))
}
```
Just to explain what the above is doing: we use `getALocalSource()` to track back the given argument (using local dataflow) to its source, and then check if it's a string constant with the given property.

Note that this won't handle the case where the URL is passed in from somewhere else (i.e. as an argument to the function that contains the call to `urlopen`). If you wish to handle that case, you'll have to use global dataflow (but I guess in general these URLs will usually be in the same scope as the call).